.github/workflows/python_tests.yml

```diff
@@ -0,0 +1,196 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: Python tests
+
+on:
+  schedule:
+    - cron: '10 2 * * *'
+  push:
+    branches: ['master', 'release-*']
+    tags: 'v*'
+  pull_request:
+    branches: ['master', 'release-*']
+    tags: 'v*'
+    paths: ['sdks/python/**', 'model/**']
+
+
+env:
+  GCP_WHEELS_STAGING_PATH: "gs://${{ secrets.GCP_BUCKET }}/${GITHUB_REF##*/}/${GITHUB_SHA}-${GITHUB_RUN_ID}/"
+  GCP_BUCKET: ${{ secrets.GCP_BUCKET }}  # beam-wheels-staging
+  GCP_PROJECT_ID: ${{ secrets.GCP_PROJECT_ID }}  # apache-beam-testing
+  GCP_REGION: ${{ secrets.GCP_REGION}}  # us-central
+
+jobs:
+
+  build_python_sdk_source:
+    name: 'Build Python SDK Source'
+    if: github.repository_owner == 'apache' && (github.event_name == 'push' || github.event_name == 'schedule')
```

Comment:
Given the size of this run, maybe we can limit this to 'schedule' only. Do you know how long it takes to run this?

Reply:
Current schedule is once a day as per line 22.  How about we run it every 4 hours?

Reply:
Does github impose any limits on how many workflow could run in parallel?

Reply:
Full workflow run takes approximately 30min. check on PR ~15 minutes.
IMHO It is not short, but also not too long. 
If you thinks it is still too long, maybe we could keep schedule and push to master/release branch events only? WDYT?

Reply:
> Does github impose any limits on how many workflow could run in parallel?

Limits for concurrent jobs are between 20-180 depends on the plan ([docs](https://docs.github.com/en/actions/getting-started-with-github-actions/about-github-actions#usage-limits)).
I heard that apache has Enterprise plan fot gh-actions but I am not sure about it. Do you have more information about it @gmcdonald ?